base/array.jl

```diff
@@ -2467,3 +2467,192 @@ end
 
 intersect(itr, itrs...) = _shrink(intersect!, itr, itrs)
 setdiff(  itr, itrs...) = _shrink(setdiff!, itr, itrs)
+
+map_unrolled(call, variables::Tuple{}) = ()
+map_unrolled(call, variables) =
+    call(first(variables)), map_unrolled(call, tail(variables))...
+
+map_unrolled(call, variables1::Tuple{}, variables2::Tuple{}) = ()
+map_unrolled(call, variables1, variables2) =
+    call(first(variables1), first(variables2)),
+    map_unrolled(call, tail(variables1), tail(variables2))...
+
+partial_map(call, fixed, variables::Tuple{}) = ()
+partial_map(call, fixed, variables) =
+    call(fixed, first(variables)), partial_map(call, fixed, tail(variables))...
```

Comment:
Didn't using a closure with map work?

Reply:
#31909

Reply:
```juila
julia> partial_map(f, c, xs) = map(x -> f(c, x), xs)
partial_map (generic function with 1 method)

julia> demo() = partial_map(+, rand(Bool) ? 1 : missing, (1, 2))
demo (generic function with 1 method)

julia> @code_warntype demo()
Variables
  #self#::Core.Compiler.Const(demo, false)
  @_2::Union{Missing, Int64}

Body::Union{Tuple{Missing,Missing}, Tuple{Int64,Int64}}
1 ─ %1 = Main.rand(Main.Bool)::Bool
└──      goto #3 if not %1
2 ─      (@_2 = 1)
└──      goto #4
3 ─      (@_2 = Main.missing)
4 ┄ %6 = @_2::Union{Missing, Int64}
│   %7 = Core.tuple(1, 2)::Core.Compiler.Const((1, 2), false)
│   %8 = Main.partial_map(Main.:+, %6, %7)::Union{Tuple{Missing,Missing}, Tuple{Int64,Int64}}
└──      return %8
```

Reply:
As long as #31909 is open I'd like to avoid closures